### ISSUE TITLE
ES6 const definitions result in "..is defined but never used no-unused-vars" error

### ISSUE THREAD

**Issue by infolock** [eslint/eslint#2780]
> If this has already been reported in another thread, my apologies for the duplicate report.

When requiring in a library via const definitions, eslint will throw the following kind of error:

`7:6 error $ is defined but never used no-unused-vars`

The code looks like this:
```js
const React = require('react');
const $ = require('jquery');
//....
```

This will fail with the error reported above when attemping to run lint. However, this will pass:
```js
const React = require('react');
var $ = require('jquery');
```

Simply changing `const` to `var` fixes the issue.  It's entirely possible I'm missing something here.  Any insight would be greatly appreciated.

```
$ npm list | grep -i eslint
|- eslint@0.20.0
|- eslint-plugin-react@2.5.0
|- grunt-eslint@12.0.0
```

**estlintConfig**
```json
{
    "globals": {
        "require": true,
        "window": true,
        "module": true
    },
    "rules": {
        "quotes": 0,
        "strict": 0,
        "no-underscore-dangle": 0,
        "new-cap": 0,
        "no-mixed-spaces-and-tabs": 0,
        "no-multi-spaces": 0,
        "react/display-name": 0,
        "react/jsx-boolean-value": 1,
        "react/jsx-quotes": 1,
        "react/jsx-no-undef": 1,
        "react/jsx-sort-props": 0,
        "react/jsx-sort-prop-types": 0,
        "react/jsx-uses-react": 1,
        "react/jsx-uses-vars": 1,
        "react/no-did-mount-set-state": 1,
        "react/no-did-update-set-state": 1,
        "react/no-multi-comp": 1,
        "react/no-unknown-property": 1,
        "react/prop-types": 1,
        "react/react-in-jsx-scope": 1,
        "react/self-closing-comp": 1,
        "react/wrap-multilines": 1
    },
    "plugins": [ "react" ],
    "ecmaFeatures": {
        "jsx": true,
        "arrowFunctions": true,
        "blockBindings": true,
        "classes": true,
        "templateStrings": true,
        "objectLiteralShorthandProperties": true
    }
}
```

**Comment by hzoo** [comment]
I think it should error for both const/var since yeah you aren't "using" it? Maybe try updating to latest eslint?

**Comment by mysticatea** [comment]
Maybe, I fixed this at https://github.com/eslint/eslint/commit/9d79da2af7c0e667ed28c33219667520d59e4d65, so please try using `eslint >= 0.22.0`.

**Comment by infolock** [comment]
@hzoo wow.... why didn't I think of that?  :confounded:

@mysticatea thanks a lot!  I will check that revision out and see how that will affect it.

**Comment by infolock** [comment]
@mysticatea I tried `0.22.0` but no dice.  Moved up to `0.23.0` and then everything is beautiful once again in the world.  Thanks!
